jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -371,15 +394,25 @@ private void buildImages() {
     results.builtImagesAndBaseImages =
         executorService.submit(
             () -> {
+              // TODO: ideally, progressDispatcher should be closed at the right moment, after the
+              // scheduled threads have completed. However, it can be tricky and cumbersome to track
+              // completion, so it may just be better to delay closing until everything ends. At
+              // least, we must ensure that it's not closed prematurely. (Garbage collection doesn't
+              // auto-close it wit the current implementation.)
+              ProgressEventDispatcher progressDispatcher =
+                  childProgressDispatcherFactory.create(
+                      "scheduling building manifests", results.baseImagesAndLayers.get().size());
+
               Map<Future<Image>, Image> builtImagesAndBaseImages = new HashMap<>();
               for (Map.Entry<Image, List<Future<PreparedLayer>>> entry :
                   results.baseImagesAndLayers.get().entrySet()) {
+                Factory progressDispatcherFactory = progressDispatcher.newChildProducer();
```

Comment:
`ProgressEventDispatcher.Factory progress... = ...`

Error Prone usually complains if a subclass name is too generic.